Sources/BugsnagPerformance/Private/Instrumentation/NetworkInstrumentation/BSGPerformanceSharedSessionProxy.mm

```diff
@@ -0,0 +1,58 @@
+//
+//  BSGPerformanceSharedSessionProxy.mm
+//  BugsnagPerformance-iOS
+//
+//  Created by Robert B on 24/10/2024.
+//  Copyright © 2024 Bugsnag. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "BSGPerformanceSharedSessionProxy.h"
+#import <objc/runtime.h>
+
+@interface BSGPerformanceSharedSessionProxy ()
+
+@property (nonatomic, strong) NSURLSession *session;
+
+@end
+
+@implementation BSGPerformanceSharedSessionProxy
+
++ (BOOL)respondsToSelector:(SEL)aSelector {
+    return [NSURLSession respondsToSelector:aSelector];
+}
+
++ (Class)class {
+    return [NSURLSession class];
+}
+
++ (BOOL)selectorShouldBeForwarded:(SEL)aSelector {
+    NSString *selectorString = NSStringFromSelector(aSelector);
+    return !([selectorString isEqual:@"finishTasksAndInvalidate"] || [selectorString isEqual:@"invalidateAndCancel"]);
```

Comment:
Let's use https://developer.apple.com/documentation/objectivec/1418736-sel_isequal here since it's faster than unwrapping and string comparing.